### DISCUSSION TITLE
[Feature]: Ask for URL first on a separate Screen and after that for Authentication

### DISCUSSION THREAD

**Discussion by akoyaxd** [immich-app/immich#1595] [Feature Request]
### Feature detail

I think the UX would be dramatically improved if the user is first asked for the Server fist and gets possible ways of authentication shown afterwards.

Currently it's very confusing for people signing in to an Instance, that only supports OIDC Login since the user gets prompted to enter his credentials in the unused fields first and only after confirming the Server URL field he sees a potential OIDC Login.

The Nextcloud Android App solved this quite nice to offer a reference.

### Platform

Mobile App

**Comment by TonioDF** [comment]
Hello, I share the same opinion, currently I use immich with oauth2 from authentik, from the web it redirects me perfectly, but from the Android app it gives me the 2 login options, with which I can login without redirecting to authentik.  So, the URL should be requested first and if you have it redirected with oauth2, automatically have the redirection and not show the immich default login option
![Screenshot_2023-02-05-21-05-36-72_d8aea95ef633eb6b7eb1cae13c34b40e](https://user-images.githubusercontent.com/91021876/218195777-8fa8cb61-1680-4892-a32a-5dfa29385972.jpg)

**Comment by jrasm91** [comment]
The mobile app has been reworked to prompt for the url first. There is another issue specifically for hiding password sign in when it is disabled.

**Reply by TonioDF** [reply]
Thank you so much. Well yes, I see that once the url is entered it continues to give the option with password, ideally it would be redirected in my case to authentik, which is where I have 2fa. But I repeat, thank you for your work

**Reply by jrasm91** [reply]
Yeah, I want it to! One of the mobile developers will have to add it. There's an open discussion specifically for that part: #2206

**Reply by TonioDF** [reply]
Great, thanks for letting me know.
